### PR TITLE
feat: handle failure scenario in mailto handling

### PR DESCRIPTION
**Pull request by acezard**
In case the user didn't have any mailing app, we weren't handling anything. This has been fixed.

This PR also includes several multiples improvements discovered while developing the feature

### CODE REVIEW

File: src/screens/error/ErrorScreen.tsx [hunk 1]
```diff
@@ -1,30 +1,63 @@
...
+  event: WebViewMessageEvent
+): Promise<undefined[]> =>
+  await Promise.all(
+    Object.keys(handlers).map(async eventId => {
+      if (!event.nativeEvent.data.includes(eventId)) return undefined
+
+      try {
+        await handlers[eventId]()
+      } catch (error) {
+        log.error('Error while handling event', eventId, error)
```
Comment (Ldoppea) [inline src/screens/error/ErrorScreen.tsx:49]: Are we sure we want to silently catch ALL events? Here our need is to silently catch only the `mailto` one.

File: src/hooks/useAppBootstrap.js [hunk 2]
```diff
@@ -140,7 +140,11 @@ export const useAppBootstrap = client => {
 
         setInitialRoute({
           stack: isHome ? fallback : undefined,
-          root: !isHome ? fallback : undefined
+          root: isConnected
+            ? !isHome
+              ? fallback
+              : undefined
+            : strings.errorScreens.offline
```
Comment (Ldoppea) [inline src/hooks/useAppBootstrap.js:147]: I don't understand why this is needed (and also why it isn't needed for the `stack` prop). Can you describe a scenario in the commit message?

File: src/screens/error/ErrorScreen.tsx (hunk 1 above)
Comment (acezard) [reply src/screens/error/ErrorScreen.tsx]: Updated so now the event has to declare if it wants to be caught or not. I set `backButton` to `false` and `mailto` to `true`

File: src/hooks/useAppBootstrap.js (hunk 2 above)
Comment (acezard) [reply src/hooks/useAppBootstrap.js:147]: This is once more an issue with app bootstrap complexity. This provides us initial error param as offline, whereas before it was handled with a race condition on the handleOffline() function. My main objective in terms of refactoring is to tackle this file and make initial routes understandable.

File: src/screens/error/ErrorScreen.tsx (hunk 1 above)
Comment (Ldoppea) [reply src/screens/error/ErrorScreen.tsx]: Why not just moving the try-catch in the handler that needs it?

**Review by Ldoppea — APPROVED** [review]

**Pull request merged**